kubernetes-itests/src/test/java/io/fabric8/kubernetes/TypedCustomResourceIT.java

```diff
@@ -0,0 +1,239 @@
+      .until(() -> petClient.inNamespace(currentNamespace).withName("pet-update").get() != null);
+    Pet updatedPet = petClient.inNamespace(currentNamespace).withName("pet-update").edit(pet1 -> {
+      pet1.getMetadata().setAnnotations(Collections.singletonMap("first", "1"));
+      return pet1;
+    });
+
+    // Then
+    assertPet(updatedPet, "pet-update", "Pigeon", null);
+    assertNotNull(updatedPet.getMetadata().getAnnotations());
+    assertEquals(1, updatedPet.getMetadata().getAnnotations().size());
+    assertEquals("1", updatedPet.getMetadata().getAnnotations().get("first"));
+  }
+
+  @Test
+  public void updateStatusSubresource() {
+    // Given
+    Pet pet = getNewPet("pet-updatestatus", "Pigeon", null);
+    PetStatus petStatusToUpdate = new PetStatus();
+    petStatusToUpdate.setCurrentStatus("Sleeping");
+
+    // When
+    pet = petClient.inNamespace(currentNamespace).create(pet);
+    await().atMost(5, TimeUnit.SECONDS)
+      .until(() -> petClient.inNamespace(currentNamespace).withName("pet-updatestatus").get() != null);
+    pet.setStatus(petStatusToUpdate);
+    Pet updatedPet = petClient.inNamespace(currentNamespace).updateStatus(pet);
+
+    // Then
+    assertPet(updatedPet, "pet-updatestatus", "Pigeon", "Sleeping");
+  }
+
+  @Test
+  public void watch() throws InterruptedException {
+    // Given
+    Pet pet = getNewPet("pet-watch", "Hamster", null);
+
+    // When
+    CountDownLatch creationEventReceived = new CountDownLatch(1);
+    Watch petWatch = petClient.inNamespace(currentNamespace).watch(new Watcher<Pet>() {
+      @Override
+      public void eventReceived(Action action, Pet resource) {
+        if (resource.getMetadata().getName().equals("pet-watch")) {
+          creationEventReceived.countDown();
+        }
+      }
+
+      @Override
+      public void onClose(WatcherException cause) { }
+    });
+    petClient.inNamespace(currentNamespace).createOrReplace(pet);
+
+    // Then
+    assertTrue(creationEventReceived.await(1, TimeUnit.SECONDS));
+    petWatch.close();
+  }
+
+  @Test
+  public void delete() {
+    // Given
+    Pet pet = getNewPet("pet-delete", "Cow", "Eating");
+
+    // When
+    petClient.inNamespace(currentNamespace).create(pet);
+    Boolean isDeleted = petClient.inNamespace(currentNamespace).withName("pet-delete").delete();
+
+    // Then
+    assertNotNull(isDeleted);
+    assertTrue(isDeleted);
+  }
+
+  private void assertPet(Pet pet, String name, String type, String currentStatus) {
+    assertNotNull(pet);
+    assertEquals(name, pet.getMetadata().getName());
+    assertEquals(type, pet.getSpec().getType());
+    if (currentStatus != null) {
+      assertEquals(currentStatus, pet.getStatus().getCurrentStatus());
+    }
+  }
+
+  @AfterClass
+  public static void cleanup() {
+    ClusterEntity.remove(TypedCustomResourceIT.class.getResourceAsStream("/pet-crd.yml"));
+  }
+
+  private Pet getNewPet(String name, String type, String currentStatus) {
```

Comment:
nit: You are not getting a pet from a store, you are creating a new Pet from scratch `createNewPet`.